src/vmm/src/devices/virtio/block/virtio/request.rs

```diff
@@ -371,14 +371,35 @@ impl Request {
         block_metrics: &BlockDeviceMetrics,
     ) -> ProcessingResult {
         let pending = self.to_pending_request(desc_idx);
+        let start_request_us = utils::time::get_time_us(utils::time::ClockType::Monotonic);
         let res = match self.r#type {
             RequestType::In => {
-                disk.file_engine
-                    .read(self.offset(), mem, self.data_addr, self.data_len, pending)
+                let res = disk.file_engine.read(
```

Comment:
if we implement record_latency_metrics  for ShareIncMetrics like we did for LatencyAggregateMetrics [here](https://github.com/sudanl0/firecracker/blob/be25214c564812babbc614ad2e4077afd0bba4a3/src/vmm/src/logger/metrics.rs#L755) we can avoid the new function update_inc_metric_with_elapsed_time.
We also don't need to remember to call the 2nd lines since drop will take care of that.
Code would look something like below:
```
let pending = self.to_pending_request(desc_idx);
        let res = match self.r#type {
            RequestType::In => {
                let _metric = block_metrics.read_duration_us.record_latency_metrics(); >> this takes care of updating the read_duration_us metric as part of dropping _metrics
                disk.file_engine
                    .read(self.offset(), mem, self.data_addr, self.data_len, pending)
            }
            RequestType::Out => {
                let _metric = block_metrics.write_duration_us.record_latency_metrics(); >> this takes care of updating the write_duration_us metric as part of dropping _metrics
                disk.file_engine
                    .write(self.offset(), mem, self.data_addr, self.data_len, pending)
            }
```

Reply:
made use of LatencyAggregateMetrics, thanks.